logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -56,34 +63,51 @@ def receive(event)
   def worker_setup
     if @workers == 1
       @worker_plugins = [self]
-      @worker_threads = []
     else
-      define_singleton_method(:handle, method(:handle_worker))
-      @worker_queue = SizedQueue.new(20)
+      define_singleton_method(:multi_handle, method(:handle_worker))
```

Comment:
Because otherwise it would affect the workers. This is just redifining it in the parent.